### PR TITLE
Add README for .NET

### PR DESCRIPTION
**Pull request by alexandrevicenzi**
This is related to #993.

### CODE REVIEW

File: src/dotnet/README.md.j2 [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+{%- endif %}
+
+## Notice
+
+The .NET packages contained in this image come from a 3rd-party repository at
+[packages.microsoft.com](https://packages.microsoft.com).
+
+You can find the respective source code in [github.com/dotnet](https://github.com/dotnet).
+
+SUSE does not provide any support or warranties for this image.
```
Comment (dcermak) [inline src/dotnet/README.md.j2:31]:
```suggestion
SUSE does not provide any support or warranties for the 3rd-party components in this image.
```

File: src/dotnet/README.md.j2 [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+# uncomment to run as non-root user
+# USER $APP_UID
+
+ENTRYPOINT ["./dotnetapp"]
+```
+
+Build and run the container image:
+
+```ShellSessionpodmanbuild -t my-dotnet-app .
+podman run -it --rm my-dotnet-app
```
Comment (dcermak) [inline src/dotnet/README.md.j2:64]:
```suggestion
```ShellSession
podman build -t my-dotnet-app .
podman run -it --rm my-dotnet-app
```

File: src/dotnet/README.md.j2 [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+```
+
+There are situations, where you don't want to run an application inside a container.
+
+To compile the application, without running it inside a container instance, use the following command:
```
Comment (dcermak) [inline src/dotnet/README.md.j2:69]:
```suggestion
To compile the application, without running it inside a container, use the following command:
```

File: src/dotnet/README.md.j2 [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+To create a self-signed certificate for testing, use the following command:
+
+```ShellSession
+podman run --rm -it -v "$PWD/https":/https:Z registry.suse.com/bci/dotnet-sdk:{{ image.version }} dotnet dev-certs https -ep /https/aspnetapp.pfx -p <PASSWORD>
+```
+
+To use a certificate and run the container image with ASP.NET Core configured for HTTPS in development or production, use the following command:
+
+```ShellSession
+podman run --rm -it -p 8081:8081 -e ASPNETCORE_HTTPS_PORTS=8081 -e ASPNETCORE_Kestrel__Certificates__Default__Password="<PASSWORD>" -e ASPNETCORE_Kestrel__Certificates__Default__Path=/https/aspnetapp.pfx -v "$PWD/https":/https:Z my-apsnet-app
```
Comment (dcermak) [inline src/dotnet/README.md.j2:164]:
```suggestion
podman run --rm -it -p 8081:8081 \
    -e ASPNETCORE_HTTPS_PORTS=8081 \
    -e ASPNETCORE_Kestrel__Certificates__Default__Password="<PASSWORD>" \
    -e ASPNETCORE_Kestrel__Certificates__Default__Path=/https/aspnetapp.pfx \
    -v "$PWD/https":/https:Z my-apsnet-app
```

File: src/dotnet/README.md.j2 [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+
+## HTTPS and certificates
+
+ASP.NET Core uses [HTTPS by default](https://docs.microsoft.com/aspnet/core/security/enforcing-ssl).
+You need a valid certificate for production deployments.
+
+To create a self-signed certificate for testing, use the following command:
+
+```ShellSession
+podman run --rm -it -v "$PWD/https":/https:Z registry.suse.com/bci/dotnet-sdk:{{ image.version }} dotnet dev-certs https -ep /https/aspnetapp.pfx -p <PASSWORD>
```
Comment (dcermak) [inline src/dotnet/README.md.j2:158]:
```suggestion
podman run --rm -it -v "$PWD/https":/https:Z \
    registry.suse.com/bci/dotnet-sdk:{{ image.version }} \
    dotnet dev-certs https -ep /https/aspnetapp.pfx -p <PASSWORD>
```

File: src/dotnet/README.md.j2 [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+```ShellSession
+podman run --rm -it -p 8081:8081 -e ASPNETCORE_HTTPS_PORTS=8081 -e ASPNETCORE_Kestrel__Certificates__Default__Password="<PASSWORD>" -e ASPNETCORE_Kestrel__Certificates__Default__Path=/https/aspnetapp.pfx -v "$PWD/https":/https:Z my-apsnet-app
+```
+{%- endif %}
+
+## Globalization
+
+.NET includes various [globalization](https://learn.microsoft.com/dotnet/core/extensions/globalization-and-localization) capabilities, including support for processing natural language text, calendars, currency, and timezones. The .NET implementation for these capabilities comes primarily from system libraries present in the container image, such as [International Components for Unicode (ICU)](https://icu.unicode.org/) and [tzdata](https://wikipedia.org/wiki/Tz_database).
+
+A good practice is to pass timezone information into a container via environment variable.
```
Comment (dcermak) [inline src/dotnet/README.md.j2:172]:
```suggestion
A good practice is to pass timezone information into a container via the environment variable `TZ`:
```

**Review by dcermak — CHANGES_REQUESTED** [review]
The readme is missing the standard footer & header.

Also, would it make it sense to have separate templates for the images? The current template looks more like three templates with a bit of common content.

File: src/dotnet/updater.py [hunk 7]
```diff
@@ -408,7 +398,7 @@ def _is_latest_dotnet(version: _DOTNET_VERSION_T, os_version: OsVersion) -> bool
                 version=ver,
                 name="dotnet-runtime",
                 is_sdk=False,
-                pretty_name=f".NET {ver} runtime",
+                pretty_name=f".NET Runtime {ver}",
```
Comment (dirkmueller) [inline src/dotnet/updater.py:401]: we use sentence style capitalization here and "runtime" suffix is used elsewhere. why do we need to change this?

File: src/dotnet/updater.py [hunk 8]
```diff
@@ -432,7 +422,7 @@ def _is_latest_dotnet(version: _DOTNET_VERSION_T, os_version: OsVersion) -> bool
                 os_version=os_version,
                 name="dotnet-aspnet",
                 is_sdk=False,
-                pretty_name=f"ASP.NET {ver} runtime",
+                pretty_name=f"ASP.NET Core Runtime {ver}",
```
Comment (dirkmueller) [inline src/dotnet/updater.py:425]: we use sentence style capitalization here

File: src/dotnet/README.md.j2 [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+
+```ShellSession
+podman build -t my-dotnet-app .
+podman run -it --rm my-dotnet-app
+```
+{%- elif image.name == "dotnet-aspnet" -%}
+To compile and deploy an application, copy the sources and build the binary:
+
+```Dockerfile
+FROM registry.suse.com/bci/dotnet-sdk:{{ image.version }} AS build
```
Comment (dirkmueller) [inline src/dotnet/README.md.j2:118]: can we avoid hardcoding the registry reference here?

**Review by dirkmueller — CHANGES_REQUESTED** [review]

File: src/dotnet/updater.py (hunk 7 above)
Comment (alexandrevicenzi) [reply src/dotnet/updater.py:401]: these are the official names and case styling used by Microsoft.

File: src/dotnet/updater.py (hunk 8 above)
Comment (alexandrevicenzi) [reply src/dotnet/updater.py:425]: these are the official names and case styling used by Microsoft.

File: src/dotnet/README.md.j2 (hunk 9 above)
Comment (alexandrevicenzi) [reply src/dotnet/README.md.j2:118]: sure, what is the best approach?

**Comment by alexandrevicenzi** [comment]
> The readme is missing the standard footer & header.

Yes, because .NET does not belong in bci_build. I have no way of reusing templates from there and I would need to duplicate it, create some hackery, or move .NET into bci_build as a package like others.

What do you suggest @dcermak @dirkmueller?

> Also, would it make it sense to have separate templates for the images? The current template looks more like three templates with a bit of common content.

Yes, it could be split, but again, it needs a bit more hackery to get it working, unless it becomes a bci_build package.

File: src/dotnet/updater.py (hunk 8 above)
Comment (alexandrevicenzi) [reply src/dotnet/updater.py:425]: See this for example https://dotnet.microsoft.com/en-us/download/dotnet/8.0

I don't see a reason to use lowercase on the package name, we don't call Apache tomcat, ASP.NET Core Runtime is the entire name, so the image would still be "ASP.NET Core Runtime container image".

**Comment by dcermak** [comment]
> > The readme is missing the standard footer & header.
> 
> Yes, because .NET does not belong in bci_build. I have no way of reusing templates from there and I would need to duplicate it, create some hackery, or move .NET into bci_build as a package like others.
> 
> What do you suggest @dcermak @dirkmueller?

I haven't tried this, but I _think_ the following should work:
- move the template to `src/bci_build/package/dotnet-sdk/README.md.j2`
- symlink `src/bci_build/package/dotnet-{runtime,aspnet}/README.md.j2` to the first template

then the standard readme reading code should pick it up **and** you should have access to additional footer & headers.

> 
> > Also, would it make it sense to have separate templates for the images? The current template looks more like three templates with a bit of common content.
> 
> Yes, it could be split, but again, it needs a bit more hackery to get it working, unless it becomes a bci_build package.

I don't think that you need to have it in the `bci_build` package for the template reader to pick it up.

File: src/dotnet/README.md.j2 (hunk 9 above)
Comment (alexandrevicenzi) [reply src/dotnet/README.md.j2]: I changed to use the `registry` variable.

**Comment by alexandrevicenzi** [comment]
@dcermak no hackery needed. It seems that the template is picked from the correct place. I also split into multiple templates and created a few partials to be imported where text was duplicated.

File: src/dotnet/updater.py (hunk 7 above)
Comment (alexandrevicenzi) [reply src/dotnet/updater.py:399]: @dmpop or @lvicoun can you advise what is best here?

File: src/dotnet/updater.py (hunk 8 above)
Comment (alexandrevicenzi) [reply src/dotnet/updater.py:423]: @dmpop or @lvicoun can you advise what is best here?

File: src/bci_build/package/dotnet-aspnet/README.md.j2 [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+
+{% include 'badges.j2' %}
+
+## Description
+
+{% include 'dotnet_description.j2' %}
+
+This image contains the ASP.NET Core and .NET runtimes and libraries,
+and is optimized for running ASP.NET Core appplications in production.
+
```
Comment (dmpop) [inline src/bci_build/package/dotnet-aspnet/README.md.j2:11]:
```suggestion
This image contains the ASP.NET Core and .NET runtimes and libraries,
and it is optimized for running ASP.NET Core applications in production.

```

File: src/bci_build/package/dotnet-runtime/README.md.j2 [hunk 11]
```diff
@@ -0,0 +1,51 @@
+# {{ image.pretty_name }} container image
+
+{% include 'badges.j2' %}
+
+## Description
+
+{% include 'dotnet_description.j2' %}
+
+This image contains the .NET runtimes and libraries and is optimized for running .NET applications in production.
```
Comment (dmpop) [inline src/bci_build/package/dotnet-runtime/README.md.j2:9]:
```suggestion
This image contains the .NET runtimes and libraries, and it is optimized for running .NET applications in production.
```

File: src/bci_build/package/dotnet-sdk/README.md.j2 [hunk 12]
```diff
@@ -0,0 +1,69 @@
+# {{ image.pretty_name }} container image
+
+{% include 'badges.j2' %}
+
+## Description
+
+{% include 'dotnet_description.j2' %}
+
+This image contains the .NET SDK, which consists of three parts:
```
Comment (dmpop) [inline src/bci_build/package/dotnet-sdk/README.md.j2:9]:
```suggestion
This image contains the .NET SDK consisting of three parts:
```

File: src/bci_build/package/dotnet-sdk/README.md.j2 [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+
+- .NET CLI
+- .NET Runtime
+- ASP.NET Core Runtime
+
+Use this image for development, build and test if .NET and ASP.NET Core applications.
```
Comment (dmpop) [inline src/bci_build/package/dotnet-sdk/README.md.j2:15]:
```suggestion
You can use the image to develop, build, and test if .NET and ASP.NET Core applications.
```

File: src/bci_build/package/dotnet-sdk/README.md.j2 [hunk 14]
```diff
@@ -0,0 +1,69 @@
...
+Build and run the container image:
+
+```ShellSession
+podmanbuild -t my-dotnet-app .
+podman run -it --rm my-dotnet-app
+```
+
+There are situations, where you don't want to run an application inside a container.
+
+To compile the application, without running it inside a container, use the following command:
```
Comment (dmpop) [inline src/bci_build/package/dotnet-sdk/README.md.j2:55]:
```suggestion
To compile the application without running it inside a container, use the following command:
```

File: src/bci_build/package/templates/dotnet_globalization.j2 [hunk 15]
```diff
@@ -0,0 +1,9 @@
+## Globalization
+
+.NET includes various [globalization](https://learn.microsoft.com/dotnet/core/extensions/globalization-and-localization) capabilities, including support for processing natural language text, calendars, currency, and timezones. The .NET implementation for these capabilities comes primarily from system libraries present in the container image, such as [International Components for Unicode (ICU)](https://icu.unicode.org/) and [tzdata](https://wikipedia.org/wiki/Tz_database).
```
Comment (dmpop) [inline src/bci_build/package/templates/dotnet_globalization.j2:3]:
```suggestion
.NET includes [globalization](https://learn.microsoft.com/dotnet/core/extensions/globalization-and-localization) capabilities, including support for processing natural language text, calendars, currency, and timezones. The .NET implementation for these capabilities is based on system libraries available in the container image, such as [International Components for Unicode (ICU)](https://icu.unicode.org/) and [tzdata](https://wikipedia.org/wiki/Tz_database).
```

File: src/bci_build/package/templates/dotnet_globalization.j2 [hunk 16]
```diff
@@ -0,0 +1,9 @@
...
+
+A good practice is to pass timezone information into a container via environment variable `TZ`.
```
Comment (dmpop) [inline src/bci_build/package/templates/dotnet_globalization.j2:5]:
```suggestion
It's consider a good practice to pass timezone information into a container via environment variable `TZ`.
```

File: src/bci_build/package/templates/dotnet_notice.j2 [hunk 17]
```diff
@@ -0,0 +1,8 @@
+## Notice
+
+The .NET packages contained in this image come from a 3rd-party repository at
+[packages.microsoft.com](https://packages.microsoft.com).
```
Comment (dmpop) [inline src/bci_build/package/templates/dotnet_notice.j2:4]:
```suggestion
The .NET packages in the image come from a third-party repository 
[packages.microsoft.com](https://packages.microsoft.com).
```

File: src/bci_build/package/templates/dotnet_notice.j2 [hunk 18]
```diff
@@ -0,0 +1,8 @@
...
+
+You can find the respective source code in [github.com/dotnet](https://github.com/dotnet).
```
Comment (dmpop) [inline src/bci_build/package/templates/dotnet_notice.j2:6]:
```suggestion
The source code is available in [github.com/dotnet](https://github.com/dotnet).
```

File: src/bci_build/package/templates/dotnet_notice.j2 [hunk 19]
```diff
@@ -0,0 +1,8 @@
...
+
+SUSE does not provide any support or warranties for the 3rd-party components in this image.
```
Comment (dmpop) [inline src/bci_build/package/templates/dotnet_notice.j2:8]:
```suggestion
SUSE does not provide any support or warranties for the third-party components in the image.
```

File: src/dotnet/updater.py (hunk 8 above)
Comment (dmpop) [reply src/dotnet/updater.py:423]: Easy: follow the capitalization used in the official product name. :smile:

**Review by dcermak — APPROVED** [review]

**Pull request merged**